spi/topology/src/main/java/quarks/topology/spi/graph/TWindowImpl.java

```diff
@@ -30,4 +35,20 @@
         Aggregate<T,U,K> op = new Aggregate<T,U,K>(window, processor);
         return feeder().pipe(op); 
     }
+
+    @Override
+    public <U> TStream<U> batch(BiFunction<List<T>, K, U> batcher) {
+        batcher = Functions.synchronizedBiFunction(batcher);
+        Window<T, K, List<T>> window =
+                Windows.window(
```

Comment:
I don't think the current policy definitions and calling order needs to be changed.

I think the batch for a last(N) windows needs to have different policies. A do-nothing conents polocy and a trigger policy (post-insertion) that processes the windows when it is of size N and then call evict.